docs/usage/config-migration.md

```diff
@@ -0,0 +1,47 @@
+# Config Migration
+
+Config migration is necessary when Renovate's configuration options are updated. Users can enable automatic config migration pull requests by setting the [configMigration](./configuration-options.md#configmigration) option to `true`.
+
+While this feature is disabled by default, we strongly recommend enabling it to prevent unexpected behavior due to outdated configurations. Keeping your config up-to-date can significantly reduce debugging time for both users and maintainers when issues arise.
+
+Here are the possible scenarios related to config migration:
+
+1. **No config migration needed**
+
+   Renovate takes no action.
+
+2. **Config migration needed and enabled**
+
+   Renovate creates a Config Migration PR and adds a link to it at the top of the Dependency Dashboard issue (if enabled).
+
+3. **Config migration needed but disabled**
+
+   If config migration is required but disabled:
+
+   We add a checkbox to the Dependency Dashboard (if enabled). We call this "on-demand config migration".
+
+   The checkbox will appear in your Dependency Dashboard issue as:
+
+   ```
+   [ ] Config migration needed. Please tick this checkbox to create an automated Config Migration PR.
+   ```
+
+   - When the user checks the box, Renovate creates a config migration PR and replaces the checkbox with a link:
+
+   ```
+   Config Migration necessary. You can view the Config Migration PR here #1
+   ```
```

Comment:
Please update the section manually, so it looks like this:

### Config migration needed, but disabled

If config migration is needed, but disabled:

Renovate adds a checkbox to the Dependency Dashboard (if enabled).
This is "on-demand config migration".

The checkbox looks like this:

```
- [ ] Config migration needed. Select this checkbox to let Renovate create an automated Config Migration PR.
```

When you select the checkbox:

1. Renovate creates a config migration PR.
2. Renovate replaces the checkbox with text with a link to the Config Migration PR.

For example:

```
Config Migration needed. See Config Migration PR: #1.
```